src/WebJobs.Extensions.DurableTask/Listener/DurableTaskScaleMonitor.cs

```diff
@@ -0,0 +1,149 @@
+        private readonly Lazy<DisconnectedPerformanceMonitor> performanceMonitor;
+        private readonly EndToEndTraceHelper traceHelper;
+
+        public DurableTaskScaleMonitor(string functionId, FunctionName functionName, string hubName, string storageConnectionString, EndToEndTraceHelper traceHelper)
+        {
+            this.functionId = functionId;
+            this.functionName = functionName;
+            this.hubName = hubName;
+            this.storageConnectionString = storageConnectionString;
+            this.performanceMonitor = new Lazy<DisconnectedPerformanceMonitor>(() => new DisconnectedPerformanceMonitor(this.storageConnectionString, this.hubName));
+            this.scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{this.functionId}-DurableTaskTrigger-{this.hubName}".ToLower());
+            this.traceHelper = traceHelper;
+        }
+
+        public ScaleMonitorDescriptor Descriptor
+        {
+            get
+            {
+                return this.scaleMonitorDescriptor;
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+        }
+
+        public async Task<DurableTaskTriggerMetrics> GetMetricsAsync()
+        {
+            DurableTaskTriggerMetrics metrics = new DurableTaskTriggerMetrics();
+
+            // Durable stores its own metrics, so we just collect them here
+            PerformanceHeartbeat heartbeat = null;
+            try
+            {
+                heartbeat = await this.performanceMonitor.Value.PulseAsync();
+            }
+            catch (StorageException e)
+            {
+                string error = string.Empty;
+                var webException = e.InnerException as WebException;
+                if (webException != null && webException.Status == WebExceptionStatus.ProtocolError)
+                {
+                    string statusCode = ((HttpWebResponse)webException.Response).StatusCode.ToString();
+                    string statusDesc = ((HttpWebResponse)webException.Response).StatusDescription;
+                    error = $"Durable Functions trigger error during PulseAsync: {statusCode}: {statusDesc}";
+                }
+                else if (webException != null && webException.Status == WebExceptionStatus.NameResolutionFailure)
+                {
+                    error = $"Durable Functions trigger storage error during PulseAsync: {webException.Message}";
+                }
+                else
+                {
+                    error = e.ToString();
+                }
+
+                this.traceHelper.ExtensionWarningEvent(this.hubName, this.functionName.Name, string.Empty, error);
+            }
+
+            if (heartbeat != null)
+            {
+                metrics.PartitionCount = heartbeat.PartitionCount;
+                metrics.ControlQueueLengths = JsonConvert.SerializeObject(heartbeat.ControlQueueLengths);

```

Comment:
What are the upper bounds for the size of these array properties (queue lengths and latencies)? Ultimately these will be stored in table storage as an entity property, and the limit for a string property is 64K (doc [here](https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-the-table-service-data-model#property-types)).

Reply:
That's a good question. @cgillum is there any documentation on the max number of control queues? I believe I found [some code](https://github.com/Azure/durabletask/blob/master/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L572) that indicates # of control queues is correlated with # of partitions?

Reply:
What's the maximum number of partitions? For both of these, just figure out what the max string length would be and ensure it fits

Reply:
We discussed this over IM, but the max partition count is 16, as documented [here](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-perf-and-scale#orchestrator-scale-out).

Reply:
According to [these]() docs, Azure Storage tables use UTF-16. So event if all 16 possible control queues were the max int value, it should be bigger than ~1288 bytes. The latencies max value shouldn't be bigger than ~1536 bytes assuming the timespan doesn't get bigger than 1000's of days